membership-attribute-service/app/services/SupporterProductDataService.scala

```diff
@@ -25,8 +25,12 @@ class SupporterProductDataService(client: DynamoDbAsyncClient, table: String, ma
     DynamoFormat.xmap[Currency, String](s => Currency.fromString(s).toRight(TypeCoercionError(new Throwable("Invalid currency"))), _.iso)
   implicit val dynamoSupporterRatePlanItem: DynamoFormat[DynamoSupporterRatePlanItem] = deriveDynamoFormat
 
-  def getAttributes(identityId: String): Future[Either[String, Option[Attributes]]] =
-    getSupporterRatePlanItems(identityId).map(ratePlanItems => mapper.attributesFromSupporterRatePlans(identityId, ratePlanItems)).value
+  def getNonCancelledAttributes(identityId: String): Future[Either[String, Option[Attributes]]] = {
+    getSupporterRatePlanItems(identityId).map { ratePlanItems =>
+      val nonCancelled = ratePlanItems.filter(_.cancellationDate.isEmpty)
```

Comment:
I think it would be safer w/r potential future changes, to test that the cancellationDate is today or in the past.